aten/src/ATen/core/PythonOpRegistrationTrampoline.cpp

```diff
@@ -0,0 +1,55 @@
+#include <c10/core/impl/HermeticPyObjectTLS.h>
+#include <ATen/core/PythonOpRegistrationTrampoline.h>
+
+namespace at {
+namespace impl {
+
+std::atomic<c10::impl::PyInterpreter*> PythonOpRegistrationTrampoline::interpreter_{nullptr};
+
+void PythonOpRegistrationTrampoline::registerInterpreter(c10::impl::PyInterpreter* interp) {
+  c10::impl::PyInterpreter* expected = nullptr;
+  interpreter_.compare_exchange_strong(expected, interp);
+  if (expected != nullptr) {
+    // This is the second (or later) Python interpreter, which means we need
+    // non-trivial hermetic PyObject TLS
+    c10::impl::HermeticPyObjectTLS::init_state();
+  }
+}
+
+void PythonOpRegistrationTrampoline::operator()(const c10::OperatorHandle& op, c10::DispatchKeySet dispatch_keys, torch::jit::Stack* stack) {
+  // The simple implementation of Python op registration involves directly
+  // registering a Python callback to the dispatcher.  However, this doesn't
+  // work with torchdeploy/multipy, where there are multiple Python
```

Comment:
what is multipy? (if different from torchdeploy)

Reply:
https://github.com/pytorch/multipy/